docs/testplan/dual_tor/dual_tor_muxcable_test_plan.md

```diff
@@ -0,0 +1,164 @@
+
+The below hardware features/properties of the Y-Cable are to be tested in this test plan. The focus is going to be on the following capabilities of the Y-Cable.
+
+- Checking the read side of the Cable and establishing the read side is one of the valid values by reading the eeprom/register spec.
+- Toggling the MUX to both the sides and ensure the mux-direction/active link side matches with the side mux is toggled to by reading the eeprom/register spec.
+- Verify that the mux-direction/active link side also complies with traffic behavior. In particular the side mux is toggled to should be the only side TOR traffic reaches the NIC side. The traffic from NIC side to TOR should be braodcasted.
+- Verify that during toggle/failover number of frames dropped is minimal or zero or whatever the vendor expectation is from Y-Cable for drop count
+- Verify that number of frames received is equal to number of frames sent in both northbound/southbound traffic flows for a predefined set period of time.
+- Verify that the manual switch counter is incremented once the MUX is toggled.
+- Verify the Vendor name is correct (one of the expected values) by reading the register spec
+- Verify part number is correct by reading the register spec. The part number should be qualified by regex match of the value returned to the value vendor provides.
+- Verify with vendor name and model value, the appropriate model's import module can be imported using y_cable_vendor_mapping.py mapping logic
+- Check the fimware version is returned correctly and is one of the acceptable values suggested by vendor. The major and minor number can be qualified by a regex match with a predefined pattern
+- Ensure firmware download, upgrade and rollback works fine.
+- Check LOS/LOL fields in the cable spec and check if they are correct
+- Check Cli show muxcable status and show muxcable hwmode muxdirection are in sync
+
+These are some of the vendor specific hardware capabilities which should be tested if available.
+
+- Ensure auto-switch gets enabled by reading the register spec once auto-switch is enabled.
+- Ensure auto-switch counter is incremented in case of link down which causes an automatic switchover
+- Ensure all the link are active and up for ToR A, ToR B and NIC.
+- Ensure BER values are in accepatble range for normal traffic mode and diagnostic mode like PRBS, Loopback.
+- Ensure EYE values are in acceptable range for normal traffic mode and diagnostic mode like PRBS, Loopback
+- Ensure Voltages and Temperatures of all the MCU's vendor provides the capability to read are within accepatble range.
+
+
+
+### Testbed Setup
+
+The test can be executed on a single Tor testbed with proposed configurations that has a muxcable put in a port with driver code. For Tests that require some traffic validation will need to run on a testbed which can send trafic from NIC to ToR and vice-versa.
```

Comment:
Can you add a diagram for the testbed topology with any connections to IXIA/PTF/VMs that will be needed?